app/src/test/java/org/mozilla/focus/session/SessionCallbackProxyTest.java

```diff
@@ -36,10 +38,15 @@ public void testOnPageFinished() {
         final Session session = mock(Session.class);
         final SessionCallbackProxy proxy = new SessionCallbackProxy(session, mock(IWebView.Callback.class));
 
+        when(session.getLoading()).thenReturn(new NonNullMutableLiveData<Boolean>(Boolean.TRUE));
         proxy.onPageFinished(true);
 
-        verify(session).setLoading(false);
-        verify(session).setSecure(true);
```

Comment:
You call something on the proxy here but you do not verify after that. Later you do verify that the APIs have been called a specific amount of times - but this doesn't guarantee that it happened at the right time.

Instead of mocking the Session object and setting all the responses manually it might make sense to use an actual Session object here:
> final Session session = spy(new Session(Source.NONE, TEST_URL));

Then verify two things:
* Set loading to true and call onPageFinished -> Verify that setLoading and setSecure are called correctly
* Set loading to false and call onPageFinished -> Verify that setLoading is not called, but setSecure.

Reply:
@pocmo test modified